apps/anoma_lib/lib/tables.ex

```diff
@@ -0,0 +1,343 @@
+defmodule Anoma.Node.Tables do
+  require Logger
+
+  ############################################################
+  #                    Types                                 #
+  ############################################################
+
+  @typedoc """
+  A table is typed by a tuple where the first element is
+  the table name, and the second one is a list of keys (i.e., columns).
+  """
+  @type table_spec :: {atom(), list(atom())}
+
+  @typedoc "a list of table specs"
+  @type table_specs :: list(table_spec)
+
+  ############################################################
+  #                    Public                                #
+  ############################################################
+
+  @doc """
+  I initialize the tables for a given node id.
+  I do this by creating all tables in the mnesia storage.
+  """
+  @spec initialize_tables_for_node(String.t(), table_specs) ::
+          :ok | {:error, :failed_to_initialize_tables}
+  def initialize_tables_for_node(node_id, tables) do
+    tables
+    |> Enum.map(fn {table, fields} ->
+      {node_table_name(node_id, table), fields}
+    end)
+    |> create_tables()
+    |> case do
+      {:error, :failed_to_create_table, _table, _fields, _err} ->
+        {:error, :failed_to_initialize_tables}
+
+      tables ->
+        case :mnesia.wait_for_tables(tables, 10_000) do
+          :ok ->
+            :ok
+
+          {:timeout, _tables} ->
+            {:error, :failed_to_initialize_tables}
+        end
+    end
+  end
+
+  @doc """
+  I initialize the mnesia storage for this entire vm.
+  """
+  @spec initialize_storage() ::
+          :ok
+          | {:error, :failed_to_initialize_storage}
+          | {:error, :failed_to_create_tables}
+  def initialize_storage() do
+    configure_mnesia()
+
+    with :ok <- create_local_schema(),
+         :ok <- :mnesia.start(),
+         :ok <- init_rocksdb() do
+      :ok
+    else
+      {:error, :failed_to_create_schema, _error} ->
+        {:error, :failed_to_create_schema}
+
+      {:error, :failed_to_initialize_rocksdb} ->
+        {:error, :failed_to_initialize_rocksdb}
+
+      _error ->
+        {:error, :failed_to_initialize_storage}
+    end
+  end
+
+  @doc """
+  Given an atom as table name, I create a node-specific name based on that.
+  """
+  @spec node_table_name(String.t(), atom()) :: atom()
```

Comment:
This is more of a comment than anything, but that is reversed from what we originally used, where we do `String.to_atom("#{name}_#{node_id}")` basically

Not sure what's a better way of naming